pkg/segment/query/segquery.go

```diff
@@ -399,6 +399,11 @@ func GetNodeResultsForSegmentStatsCmd(queryInfo *QueryInformation, sTime time.Ti
 			ErrList: []error{err},
 		}
 	}
+	for _, qsr := range sortedQSRSlice {
+		if qsr.sType == structs.RAW_SEARCH || qsr.sType == structs.PQS {
+			qsr.sType = structs.SEGMENT_STATS_SEARCH
+		}
+	}
```

Comment:
I don't quite understand why you're doing this. It seems like we'd expect the sType to already be `SEGMENT_STATS_SEARCH` since we're in the function `GetNodeResultsForSegmentStatsCmd()`. Why would we get here with `RAW_SEARCH` or `PQS`?

Reply:
Yes, actually because I am sending `SQRS` that we are fetching from `searched.initializeQSRs`, and passing it to the `segmentStats` function, it will not fetch again from the `getAllSegmentsInAggs` function, which is why I have to add this code.

But now I have removed this and passing for `SegmentStatsCmd` and also only fetching the `SQRs`, if the command type is `GroupByCmd`.